monix-tail/shared/src/main/scala/monix/tail/internal/IterantMapBatch.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2014-2018 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.tail.internal
+
+import cats.effect.Sync
+import cats.syntax.all._
+import monix.execution.misc.NonFatal
+import monix.tail.Iterant
+import monix.tail.Iterant.{Halt, Last, Next, NextBatch, NextCursor, Suspend}
+import monix.tail.batches.Batch
+import monix.tail.internal.IterantUtils.signalError
+
+private[tail] object IterantMapBatch {
+  /**
+    * Implementation for `Iterant#mapBatch`
+    */
+  def apply[F[_], A, B](source: Iterant[F, A], f: A => Batch[B])
+    (implicit F: Sync[F]): Iterant[F, B] = {
+
+    def processBatch(ref: NextCursor[F, A]): NextBatch[F, B] = {
+      val NextCursor(cursor, rest, stop) = ref
+
+      val batch = if(cursor.hasNext()) f(cursor.next()) else Batch[B]()
```

Comment:
@alexandru 
I've changed it to what you suggested but I'm not sure about this case. Is it ok to handle it like this?
Alternatively I could return `Suspend` when I get empty cursor

Reply:
Sorry for not replying, I'm busy these days at work.

A `Suspend` is better than an empty `NextBatch`. Although it's not a major issue, since they should be equivalent.

The rest looks pretty good. Do this small change and it's ready to go.